Crash on export when submission is empty

#### Software versions
Briefcase v1.17.4

#### Problem description
When a submission is empty (or presumably when one is corrupt), the whole export crashes with no real hints as to what is going on.

#### Steps to reproduce the problem
1. Start a pull operation and cancel it. Verify that the last submission started is empty
1. Start an export
1. See that export files only have headers. Export details stop after some percentage.
1. Debug to see crash in `parseAttribute` 

#### Expected behavior
Canceling should probably not leave an empty submission.
If there is an empty submission, that should be logged, the submission should be skipped, and export should continue. There is a `failed_submission` file generated, but it's empty so it doesn't really help.

The actual crash is
```
java.lang.RuntimeException: java.lang.RuntimeException: Document has no root element!
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:593)
	at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:735)
	at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:496)
	at org.opendatakit.briefcase.export.ExportToCsv.export(ExportToCsv.java:110)
	at org.opendatakit.briefcase.export.ExportToCsv.export(ExportToCsv.java:62)
	at org.opendatakit.briefcase.ui.export.ExportPanel.lambda$null$7(ExportPanel.java:216)
	at org.opendatakit.briefcase.reused.job.Job.lambda$run$0(Job.java:54)
	at org.opendatakit.briefcase.reused.job.Job.lambda$thenRun$7(Job.java:123)
	at org.opendatakit.briefcase.reused.job.Job.lambda$thenRun$7(Job.java:122)
	at org.opendatakit.briefcase.reused.job.JobsRunner.lambda$null$1(JobsRunner.java:65)
	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.RuntimeException: Document has no root element!
	at org.kxml2.kdom.Document.getRootElement(Unknown Source)
	at org.opendatakit.briefcase.export.XmlElement.of(XmlElement.java:68)
	at org.opendatakit.briefcase.export.SubmissionParser.lambda$parseSubmission$5(SubmissionParser.java:130)
	at java.util.Optional.flatMap(Optional.java:241)
Caused by: java.lang.RuntimeException: Document has no root element!
```

The submission is identified as bad in `parseAttribute` but it still part of the list of submissions to export. It should not be included.

--- a/src/org/opendatakit/briefcase/export/ExportToCsv.java
+++ b/src/org/opendatakit/briefcase/export/ExportToCsv.java
@@ -37,7 +37,6 @@ import java.time.OffsetDateTime;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
-import java.util.concurrent.atomic.AtomicInteger;
 import org.bushe.swing.event.EventBus;
 import org.opendatakit.briefcase.model.FormStatus;
 import org.opendatakit.briefcase.model.form.FormMetadata;
@@ -143,14 +142,13 @@ public class ExportToCsv {
   }
 
   private static SubmissionExportErrorCallback buildParsingErrorCallback(Path errorsDir) {
-    AtomicInteger errorSeq = new AtomicInteger(1);
     // Remove errors from a previous export attempt
     if (exists(errorsDir))
       deleteRecursive(errorsDir);
     return (path, message) -> {
       if (!exists(errorsDir))
         createDirectories(errorsDir);
-      copy(path, errorsDir.resolve("failed_submission_" + errorSeq.getAndIncrement() + ".xml"));
+      copy(path, errorsDir.resolve("failed_submission_" + path.getParent().getFileName() + ".xml"));
       log.warn("A submission has been excluded from the export output due to some problem ({}). If you didn't expect this, please ask for support at https://forum.getodk.org/c/support", message);
     };
   }

--- a/src/org/opendatakit/briefcase/export/SubmissionParser.java
+++ b/src/org/opendatakit/briefcase/export/SubmissionParser.java
@@ -86,11 +86,12 @@ public class SubmissionParser {
         .forEach(instanceDir -> {
           Path submissionFile = instanceDir.resolve("submission.xml");
           try {
-            Optional<OffsetDateTime> submissionDate = readSubmissionDate(submissionFile, onParsingError);
+            Optional<OffsetDateTime> submissionDate = readSubmissionDate(submissionFile);
             paths.add(Pair.of(submissionFile, submissionDate.orElse(OffsetDateTime.of(1970, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC))));
           } catch (Throwable t) {
-            log.error("Can't read submission date", t);
+            log.error("Parse error attempting to read instance date", t);
             EventBus.publish(ExportEvent.failureSubmission(formDef, instanceDir.getFileName().toString(), t));
+            onParsingError.accept(submissionFile, "Parse error attempting to read instance date");
           }
         });
     return paths.parallelStream()
@@ -152,31 +153,25 @@ public class SubmissionParser {
     });
   }
 
-  private static Optional<OffsetDateTime> readSubmissionDate(Path path, SubmissionExportErrorCallback onParsingError) {
+  private static Optional<OffsetDateTime> readSubmissionDate(Path path) throws XMLStreamException {
     try (InputStream is = Files.newInputStream(path);
          InputStreamReader isr = new InputStreamReader(is, UTF_8)) {
-      return parseAttribute(path, isr, "submissionDate", onParsingError)
+      return parseAttribute(isr, "submissionDate")
           .map(Iso8601Helpers::parseDateTime);
     } catch (IOException e) {
       throw new CryptoException("Can't decrypt file", e);
     }
   }
 
-  private static Optional<String> parseAttribute(Path submission, Reader ioReader, String attributeName, SubmissionExportErrorCallback onParsingError) {
-    try {
-      XMLStreamReader reader = xmlInputFactory.createXMLStreamReader(ioReader);
-      while (reader.hasNext())
-        if (reader.next() == START_ELEMENT)
-          for (int i = 0, c = reader.getAttributeCount(); i < c; ++i)
-            if (reader.getAttributeLocalName(i).equals(attributeName))
-              return Optional.of(reader.getAttributeValue(i));
-    } catch (XMLStreamException e) {
-      log.error("Can't parse submission", e);
-      onParsingError.accept(submission, "parsing error");
-    }
+  private static Optional<String> parseAttribute(Reader ioReader, String attributeName) throws XMLStreamException {
+    XMLStreamReader reader = xmlInputFactory.createXMLStreamReader(ioReader);
+    while (reader.hasNext())
+      if (reader.next() == START_ELEMENT)
+        for (int i = 0, c = reader.getAttributeCount(); i < c; ++i)
+          if (reader.getAttributeLocalName(i).equals(attributeName))
+            return Optional.of(reader.getAttributeValue(i));
     return Optional.empty();
   }
-
 
   private static Optional<Submission> decrypt(Submission submission, SubmissionExportErrorCallback onError) {
     List<Path> mediaPaths = submission.getMediaPaths();

